### PR TITLE
Model Evaluator

### PR DESCRIPTION
**Pull request by Arminh388**

### CODE REVIEW

File: model/model_evaluator.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import torch
+from game import Game
+import sys
+sys.path.insert(1, '/home/armandobean/Qmind/arbitrage/utils')
```
Comment (mecsabb) [inline model/model_evaluator.py]: Should change this to be a relative path, Python imports are a such a pain lol

File: model/model_evaluator.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+def evaluate_model_on_random_graph(model: GCN):
+    try:
+        # Create a random graph for evaluation
+        graph_generator = GraphGenerator(min_weight=1, max_weight=10, max_nodes=26)
+        random_graph = graph_generator.generate_fully_connected_graph(num_nodes=5)
+
+        print("Random Graph:")
+        print(random_graph.data)
+
+        # Convert graph data to PyTorch tensors
```
Comment (mecsabb) [inline model/model_evaluator.py]: I think you can just pass the Game obj random_graph.data, should be PyG Data type already.  Lmk if I'm wrong I've become unfamiliar with out graphutils file.

File: model/model_evaluator.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        x = torch.tensor(random_graph.x, dtype=torch.float)
+        edge_index = torch.tensor(random_graph.edge_index, dtype=torch.long)
+        edge_weight = torch.tensor(random_graph.edge_attr, dtype=torch.float) if hasattr(random_graph, 'edge_attr') else None
+
+        # Create a Game instance with the random graph
+        game = Game(graph=Data(x=x, edge_index=edge_index, edge_attr=edge_weight))
+
+        # Evaluate the model on the random graph
+        p, v = model(game, edge_weight)
```
Comment (mecsabb) [inline model/model_evaluator.py]: Nice, now at the end of this function you are going to want to find the total rewards this model gets on the generated graph.  In order to do this, you're going to have to pass the Game object the edge you'd like to move across (Game.step method), and sum the reward it gets for doing so each time (returned by step).  You have to do this until it reaches a terminal state (i.e. starting node), which is represented by the is_terminal value returned by Game.step.  

Since the model's forward function (called by model(game, edge_weight)) outputs a policy for a given graph state, you'll want to pass the edge the model recommends with highest probability (i.e. argmax of policy).

File: model/model_evaluator.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+from model import GCN
+from torch_geometric.data import Data
+from graphutils import GraphGenerator
+
+
+def evaluate_model():
+    #TODO
+    pass
+
+def model_evaluator(model, best_model, num_instances=100):
```
Comment (mecsabb) [inline model/model_evaluator.py]: For now, this can run with the random graph function

File: model/model_evaluator.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+        print("Policy (p) Output:")
+        print(p)
+
+        print("Value (v) Output:")
+        print(v)
+
+    except Exception as e:
+        print(f"An error occurred during evaluation: {e}")
+
```
Comment (mecsabb) [inline model/model_evaluator.py:47]: A final recommendation I have is that this might output a lot (especially when you run it on 100 graphs), and that might get out of hand.  Take some time to think carefully about what you should output and you can also think about making log files for the large runs and just printing the most important info in terminal.

**Review by mecsabb — CHANGES_REQUESTED** [review]

**Review by mecsabb — APPROVED** [review]
Nice, I'll pick it up from here
